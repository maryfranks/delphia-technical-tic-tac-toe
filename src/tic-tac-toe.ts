export class TicTacToeGame {
    constructor() {

    }

    play() {
        console.log("Welcome!")
    }
}