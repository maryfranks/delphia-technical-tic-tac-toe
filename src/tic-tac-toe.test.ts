import { TicTacToeGame } from './tic-tac-toe' 

describe("Class: TicTacToeGame", () => {
  describe("Instance Method: displayBoard", () => {
    it("displays human readable gameBoard", () => {
      // arrange
      const newGame = new TicTacToeGame()
      jest.spyOn(console, "log")
      // act
      newGame.displayBoard()
      // assert
      // @ts-ignore:next
      expect(console.log.mock.calls[0][0]).toMatchSnapshot()
    })
  })
})

test('test', () => {
  const newGame = new TicTacToeGame()

  jest.spyOn(console, "log")

  newGame.play()
  expect(console.log).toHaveBeenCalledWith("Welcome!");
});