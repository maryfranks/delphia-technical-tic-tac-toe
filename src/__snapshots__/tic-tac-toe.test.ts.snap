// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class: TicTacToeGame Instance Method: displayBoard displays human readable gameBoard 1`] = `
"┌─────────┬────┬────┬────┐
│ (index) │ 0  │ 1  │ 2  │
├─────────┼────┼────┼────┤
│    0    │ '' │ '' │ '' │
│    1    │ '' │ '' │ '' │
│    2    │ '' │ '' │ '' │
└─────────┴────┴────┴────┘"
`;
